content/posts/2021-11-21-tăng-tốc.md

```diff
@@ -0,0 +1,65 @@
+
+### 1. Fit your need
+![npm start](/media/npm-start.jpg)
+
+It sounds obvious that you should choose a solution that solves your problem. A package usually tries to solve a set of problems in general. But sometimes, your use case is particular to your situation, and the package might not fit. Please make sure the third party code you are going to install handles sufficient, not too much, not too over engineer your problem. Many times, [DIY](#do-it-yourself) is a good option
+
+### 2. Easy to adopt
+![Easy](/media/easy.jpg)
+
+In industry, you don't work alone. So, pick a solution that your team members are able to adapt to quickly. You don't want to choose a solution that you are the only one who can implement and maintain.
+
+### 3. Popularity
+![Popularity](/media/popular.jpg)
+
+It's usually true that a good solution is a popular one. If a solution is good, why isn't it popular? I measure the popularity of a javascript package by the number of github stars and the number of weekly downloads on [https://npmjs.com](https://npmjs.com). A package with a big community proves that it solves the problem for many people. It comes with the advantages of having better support when we run into an issue. The popular package has more maintainers, contributors and users, so it's more mature than the less popular ones. It usually has good documentation and tutorials, which helps you to easily adopt and use it correctly and efficiently.
+
+**Bonus:** [https://stateofjs.com/](https://stateofjs.com/) is a great resource for you to know about the trending in javascript every year
+
+### 4. In active development
+![Team work](/media/teamwork.jpg)
+
+It's very important that's you are choosing an active project instead of a dead/ unmaintained project. The active project gets better and better over time through community feedback. An unmaintained project does not move forward and does not fix functional bugs and patch security issues. Sometimes, a very popular package could get abandoned and go into the "freeze" state with many open issues and pull requests. It's might used to a great solution in the past, but maybe this is the sign for us we have to move on. An example is [https://github.com/jamiebuilds/react-loadable](https://github.com/jamiebuilds/react-loadable). It's a great solution for a very long time to split code in React. I totally love it. But's stale now with many issues and PRs since 2018 (this post is written at the end of 2021). Now, if I need to split code in React, I rather use https://github.com/gregberge/loadable-components, which is in active development, getting more popular, patch bugs reported by the community and most importantly, solve my problems. My personal advice: choose a package that active in the last 3-6 months, with issues are being resolved and PRs are being merged.
```

Comment:
- An unmaintained project does not move forward and does not fix functional bugs and patch security issues. câu này nhiều and quá, đọc hơi mệt, nên dùng dấu phẩy => An unmaintained project does not move forward, does not fix functional bugs, and does not patch security issues.
- 2 link Github nên ghi tên lib rồi nhúng link vào như ở dưới
- It's might used to a great solution in the past => It might be a great solution in the past
- But's stale now => But it's stale now
- that active => that is active/ that has contributions